packages/config/src/bin/main.js

```diff
@@ -13,27 +15,42 @@ const resolveConfig = require('../main')
 
 const { parseFlags } = require('./flags')
 
+const pWriteFile = promisify(writeFile)
+
 // CLI entry point
 const runCli = async function () {
   try {
-    const { stable, ...flags } = parseFlags()
+    const { stable, output = DEFAULT_OUTPUT, ...flags } = parseFlags()
     const result = await resolveConfig(flags)
-    handleCliSuccess(result, stable)
+    await handleCliSuccess(result, stable, output)
   } catch (error) {
     handleCliError(error)
   }
 }
 
-// The result is printed as JSON on stdout on success (exit code 0)
-const handleCliSuccess = function (result, stable) {
+const DEFAULT_OUTPUT = '-'
+
+// The result is output as JSON on success (exit code 0)
+const handleCliSuccess = async function (result, stable, output) {
   const resultA = serializeApi(result)
   const resultB = omit(resultA, SECRET_PROPERTIES)
   const stringifyFunc = stable ? stableStringify : JSON.stringify
   const resultJson = stringifyFunc(resultB, null, 2)
-  console.log(resultJson)
+  await outputResult(resultJson, output)
   process.exitCode = 0
 }
 
+const outputResult = async function (resultJson, output) {
+  if (output === '-') {
+    console.log(resultJson)
+    return
+  }
+
+  // TODO: remove once the buildbot stops using stdout
+  console.log(resultJson)
+  await pWriteFile(output, resultJson)
```

Comment:
Do we want to use something like `mkdirp` to make sure the directory structure exists?

Reply:
Great point! Fixed.